### PR TITLE
Refactors the ParticleClaimStatement AST node.

### PR DESCRIPTION
**Pull request by csilvestrini**
This pulls out an Expression field, similar to what we did for Checks.
This makes the code reusable for datastore claims (since they don't have handles), and will be useful for building boolean expressions in claims.

Also changes the Edge.claim field from being a Claim object to being a ClaimExpression object. Edges don't really need the full Claim object (the handle field doesn't apply to them), and this will become important for datastore claims, which need to make claims about edges, but won't have a handle field.

### CODE REVIEW

File: src/runtime/manifest-parser.peg [hunk 1]
```diff
@@ -257,14 +257,14 @@ Particle
           location: location() // TODO: FIXME Get the locations of the item descriptions.
         } as AstNode.Description;
         item.description.forEach(d => description[d.name] = d.pattern || d.patterns[0]);
-      } else if (item.kind === 'particle-trust-claim-is-tag' || item.kind === 'particle-trust-claim-derives-from') {
+      } else if (item.kind === 'particle-trust-claim' || item.kind === 'particle-trust-claim') {
```
Comment (raulverag) [inline src/runtime/manifest-parser.peg]: This no longer need to check the kind twice, as they are both the same.

File: src/runtime/particle-claim.ts [hunk 2]
```diff
@@ -10,36 +10,44 @@
 
 import {HandleConnectionSpec} from './particle-spec';
 import {ParticleClaimIsTag, ParticleClaimDerivesFrom, ParticleClaimStatement} from './manifest-ast-nodes';
+import {browserLocalStorage} from '@tensorflow/tfjs-core/dist/io/local_storage';
```
Comment (raulverag) [inline src/runtime/particle-claim.ts]: Huh? Why is this here? It isn't referenced in any of your edits below.

Comment (csilvestrini) [reply src/runtime/particle-claim.ts]: Done

File: src/runtime/manifest-parser.peg (hunk 1 above)
Comment (csilvestrini) [reply src/runtime/manifest-parser.peg]: Done
